modules/network/pre-existing-vpc/main.tf

```diff
@@ -18,10 +18,24 @@
 data "google_compute_network" "vpc" {
   name    = var.network_name
   project = var.project_id
+
+  lifecycle {
+    postcondition {
+      condition     = self.name != null
```

Comment:
I assume you tested that this fails when network is not found. Assuming yes, then this is not incorrect but my preference would be to use network_self_link instead. Self link tends to be a more holistic attribute as it includes project and region. When referring to a network we tend to prefer passing the self link as opposed to the name, so asserting on the self link seems like a slightly better option.

Same for subnetwork.

Reply:
Done